belvedere/src/main/java/zendesk/belvedere/BelvedereUi.java

```diff
@@ -14,6 +20,123 @@
 
     private final static String FRAGMENT_TAG = "BelvedereDialog";
     private final static String EXTRA_MEDIA_INTENT = "extra_intent";
+    private final static String FRAGMENT_TAG_POPUP = "belvedere_image_stream";
+
+    public static ImageStreamBuilder imageStream(Context context) {
+        return new ImageStreamBuilder(context);
+    }
+
+    public static ImageStream install(AppCompatActivity activity) {
+        final FragmentManager supportFragmentManager = activity.getSupportFragmentManager();
+        Fragment fragment = supportFragmentManager.findFragmentByTag(FRAGMENT_TAG_POPUP);
+
+        ImageStream popupBackend;
+        if(fragment instanceof ImageStream) {
+            popupBackend = (ImageStream) fragment;
+        } else {
+            popupBackend = new ImageStream();
+            supportFragmentManager
+                    .beginTransaction()
+                    .add(popupBackend, FRAGMENT_TAG_POPUP)
+                    .commit();
+        }
+
+        popupBackend.setKeyboardHelper(KeyboardHelper.inject(activity));
+        return popupBackend;
+    }
+
+    public static class ImageStreamBuilder {
+
+        private final Context context;
+        private boolean resolveMedia = true;
+        private List<MediaIntent> mediaIntents = new ArrayList<>();
+        private List<MediaResult> selectedItems = new ArrayList<>();
+        private List<MediaResult> extraItems = new ArrayList<>();
+
+        private ImageStreamBuilder(Context context){
+            this.context = context;
+        }
+
+        public ImageStreamBuilder withMediaIntents(List<MediaIntent> mediaIntents) {
+            this.mediaIntents = mediaIntents;
+            return this;
+        }
+
+        public ImageStreamBuilder withMediaIntents(MediaIntent... mediaIntents) {
+            this.mediaIntents = Arrays.asList(mediaIntents);
+            return this;
+        }
+
+        public ImageStreamBuilder withCameraIntent() {
+            this.mediaIntents.add(Belvedere.from(context).camera().build());
+            return this;
+        }
+
+        public ImageStreamBuilder withDocumentIntent(String contentType, boolean allowMultiple) {
+            this.mediaIntents.add(Belvedere.from(context).document().allowMultiple(allowMultiple).contentType(contentType).build());
+            return this;
+        }
+
+        public ImageStreamBuilder withSelectedItems(List<MediaResult> mediaResults) {
+            this.selectedItems = mediaResults;
```

Comment:
Should we assign or add mediaResults?

Assigning can cause a side effect when first calling method that adds intent to list (camera, document) and then another method that reassigns mediaIntents.
For example, when first adding camera intent and then media intents, the original camera intent will be gone 
```
imageStreamBuilder.withCameraIntent().withSelectedItems(mediaResults)
```